tests/integ/core/conftest.py

```diff
@@ -0,0 +1,27 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pytest
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers", "core: subset of integ tests that must run for every pull request"
+    )
+
+
+def pytest_collection_modifyitems(config, items):
+    for item in items:
+        if "tests/integ/core" in str(item.fspath):
+            item.add_marker(pytest.mark.core)
```

Comment:
So with this we can do `tox -e py310 -- test/integ -m core`?

Idea being that we'll be able to split things up here and later on in our buildspec do

`tox -e py310 -- test/integ -m $SUB_TESTS`

where we pass `$SUB_TESTS` (or whatever we call it) as an environment variable from our github actions yml

Reply:
Yup! that's correct. 

This is applied as a filter *after* pytest performs collection and adds a marker to the tests that are in this directory.
Locally the collection runs quickly <3mins 

`pytest tests/integ --collect-only --quiet -m core`

Reply:
nicee